src/libraries/Microsoft.PowerFx.Interpreter/Marshal/ObjectMarshaller.cs

```diff
@@ -59,7 +59,7 @@ internal bool TryGetField(object source, string name, out FormulaValue fieldValu
             return false;
         }
 
-        internal IEnumerable<NamedValue> GetFields(object source)
+        protected internal virtual IEnumerable<NamedValue> GetFields(object source)
```

Comment:
>source [](http://example.com/codeflow?start=76&length=6)

I think we could actually make this non-virtual by just relying on the property enumeration from the FormulaType, and then calling Marshal() for each one.

Reply:
Having control over this list of fields is important. It allows you to expose fields that aren't properties, or expose properties from sub-classes as if they were properties on this object itself.

Reply:
ack - but that's already captured in Type information.